src/MCPClient/lib/clientScripts/verify_AIP.py

```diff
@@ -0,0 +1,244 @@
+    if file_.checksum != expected_checksum:
+        event_outcome_detail_note = (
+            'The checksum {db_checksum} for file {file_uuid} from the'
+            ' database did not match the corresponding checksum from the'
+            ' Bag manifest file {manifest_checksum}'.format(
+                file_uuid=file_.uuid,
+                db_checksum=file_.checksum,
+                manifest_checksum=expected_checksum))
+        raise VerifyChecksumsError(
+            write_premis_event(sip_uuid, checksum_type, 'Fail',
+                               event_outcome_detail_note))
+
+
+def verify_checksums(bag, sip_uuid):
+    """Verify that the checksums generated at the beginning of transfer match
+    those generated near the end of ingest by bag, i.e., "Prepare AIP"
+    (bagit_v0.0).
+    """
+    checksum_type = get_setting(
+        'checksum_type', mcpclient_settings.DEFAULT_CHECKSUM_ALGORITHM)
+    try:
+        manifest_path = get_manifest_path(bag, sip_uuid, checksum_type)
+        path2checksum = parse_manifest(manifest_path, sip_uuid, checksum_type)
+        verification_count = 0
+        for file_ in File.objects.filter(sip_id=sip_uuid):
+            if not file_.currentlocation.startswith('%SIPDirectory%objects/'):
+                continue
+            file_path = file_.currentlocation.replace('%SIPDirectory%', '', 1)
+            assert_checksum_types_match(file_, sip_uuid, checksum_type)
+            expected_checksum = get_expected_checksum(
+                file_, sip_uuid, checksum_type, path2checksum, file_path,
+                manifest_path)
+            assert_checksums_match(file_, sip_uuid, checksum_type,
+                                   expected_checksum)
+            verification_count += 1
+    except VerifyChecksumsError as err:
+        print(err)
+        return 1
+    event_outcome_detail_note = (
+        'All {verification_count} checksums generated at start of transfer'
+        ' match those generated by BagIt (bag).'.format(
+            verification_count=verification_count))
+    write_premis_event(sip_uuid, checksum_type, 'Pass',
+                       event_outcome_detail_note)
+    print(event_outcome_detail_note)
+    return 0
+
+
+def verify_aip():
+    """Verify the AIP was bagged correctly by extracting it and running
+    verification on its contents. This is also where we verify the checksums
+    now that the verifyPREMISChecksums_v0.0 ("Verify checksums generated on
+    ingest") micro-service has been removed. It was removed because verifying
+    checksums by calculating them in that MS and then having bagit calculate
+    them here was redundant.
+
+    sys.argv[1] = UUID
+      UUID of the SIP, which will become the UUID of the AIP
+    sys.argv[2] = current location
+      Full absolute path to the AIP's current location on the local filesystem
+    """
+
+    sip_uuid = sys.argv[1]  # %sip_uuid%
+    aip_path = sys.argv[2]  # SIPDirectory%%sip_name%-%sip_uuid%.7z
```

Comment:
Nice way to put it with the examples, that's useful :)

Reply:
I didn't do it, but I'll take the compliment